smithy-model/src/main/java/software/amazon/smithy/model/traits/EnumTrait.java

```diff
@@ -28,19 +28,23 @@
 /**
  * Constrains string values to one of the predefined enum constants.
  */
-public final class EnumTrait extends AbstractTrait implements ToSmithyBuilder<EnumTrait> {
+public class EnumTrait extends AbstractTrait implements ToSmithyBuilder<EnumTrait> {
```

Comment:
Can you add some docs about this trait and reference the synthetic trait variant? Maybe document that this is deprecated and to use the enum shape.